stroyprombeton/management/commands/parse_stalbeton.py

```diff
@@ -9,88 +10,119 @@
+    def __str__(self):
+        return self.path
 
-    def work_doc(self) -> typing.List[CategoryPage]:
-        # parse work doc categories
-        pass
+    @property
+    def title(self) -> str:
+        return self.soup.find('title').text
 
+    @property
+    def h1(self) -> str:
+        return self.soup.find('h1').text
 
-class CatalogPage(Page):
-    path = 'catalog'
+    @property
+    def description(self) -> str:
+        return self.soup.select_one('meta[name="Description"]')['content']
 
-    def roots(self) -> typing.List['RootCategoryPage']:
-        # parse root categories
-        pass
 
-    def second_level(self) -> typing.List['SecondLevelCategoryPage']:
-        # parse second level categories
-        pass
+class CategoryPage(Page):
+    @property
+    def text(self) -> str:
+        """
+        Only category page has unique text.
+
+        Every another text has autogenerated content.
+        """
+        return self.soup.select_one('#js-category-description').text
 
 
 class RootCategoryPage(CategoryPage):
-    pass
+    def second_level(self) -> typing.List['SecondLevelCategoryPage']:
+        return [
+            SecondLevelCategoryPage(p['href'])
+            for p in self.soup.select('h2 > a.catalog-list__link')
+        ]
 
 
 class SecondLevelCategoryPage(CategoryPage):
     def third_level(self) -> typing.List['ThirdLevelCategoryPage']:
-        # parse it
-        pass
+        return [
+            ThirdLevelCategoryPage(p['href'])
+            for p in self.soup.select('h2 > a.catalog-list__link')
+        ]
 
 
 class ThirdLevelCategoryPage(CategoryPage):
     def options(self) -> list:
-        # parse it
-        pass
+        # @todo #741:60m  Parse stalbeton's options.
+        raise NotImplemented()
+
+    # @todo #741:60m Parse series from stalbeton.
+    #  Series are already parsed as text strings.
+    #  Parse them as separated pages to get options-series relation.
+    def series(self) -> typing.List[str]:
+        return [
+            item.text for item in self.soup.select(
+                '.documentation-block span.documentation-block__item > a'
+            )
+        ]
 
 
-# @todo #736:120m  Parse stalbeton pages.
-#  See "parse it" comments inside the classes.
 def parse():
-    # @todo #736:60m  Try to use stalbeton sitemap.xml.
-    #  To get all category links. If you'll be succeed, use it
-    #  instead of nested category parsing, drafted with the code below.
-    roots = CatalogPage().roots()  # Ignore PyFlakesBear
-    second_level = CatalogPage().second_level()
-    third_level = chain.from_iterable((s.third_level() for s in second_level))
-    options = chain.from_iterable((t.options() for t in third_level))  # Ignore PyFlakesBear
-    # save it into our DB
+    main = Page(path='/')
+    through = ThroughElements(page=main)
+    roots = through.roots()
+    seconds = chain.from_iterable((r.second_level() for r in roots))
```

Comment:
We can make `CategoryPage` implementers polymorphic 
```python
class CategoryPage(Page):
    def children(self) -> List['CategoryPage']:
        raise NotImplementedError
    def options(self) -> list:
        return list(chain.from_iterable(child.options() for child in self.children()))
    def series(self) -> list:
        return list(chain.from_iterable(child.series() for child in self.children()))
```
And implement them in this way:
```python
class Categories(CategoryPage):
    def children(self):
        return list(chain.from_iterable(category.children() for category in self.categories))

class RootCategoryPage(CategoryPage):
    def children(self):
        return [SecondLevelCategoryPage(...) for ... in ...]
class SecondLevelCategoryPage(...):
class ThirdLevelCategoryPage(...): 
    def children(self):
        return []
    def options(self):
        # parse
```
and use it so:
```
categories = Categories(roots)
options = categories.options()
series = Categories(categories.children()[:100]).series()
```

Reply:
@artemiy312 , semantically not every category page has it's own options we should extract. The same is for series. I proposed a more rude non-polymorphic structure to clarify some points:
- we can correctly extract options and series only from third level categories
- about categories nesting: only roots can give us seconds. And only seconds - thirds.

Now i'm going to merge the task, but i'll create separated subtask to discuss it.

Categories class is a good idea, i'll subtask it